week-3/server.js

```diff
@@ -0,0 +1,258 @@
+        const orderDate= req.body.orderDate;
+        const orderRef=req.body.orderRef;
+        const customerId=parseInt(req.params.customerId);
+        if (customersIdArr.includes(customerId) ){
+        pool.query("insert into orders (order_date,order_reference,customer_id) "+
+        "values($1,$2,$3) ",[orderDate,orderRef,customerId ],(error,result)=> {
+            if ((error)== undefined){
+                res.send("New order added.");
+               // res.json(result.rows[0]);
+                //res.send("New customer added.");
+            }
+            else {
+            res.status(400).send("wrong request "+error)
+}
+        });
+    }
+    else
+    res.status(400).send("The  id of the customer  must be existed  ")
+    });
+    
+
+    //PUT endpoint `/customers/:customerId`
+    app.put("/customers/:customerId" ,function (req,res){
+        const customerId=parseInt( req.params.customerId);
+        const newName=req.body.name;
+        const newCity=req.body.city;
+        const newAdd=req.body.address;
+        const newCountry=req.body.country;
+        if (customersIdArr.includes(customerId) ){
+                    pool.query("update customers set name=$1 ,city =$2,address=$3 ,country=$4"+
+        "where id =$5"
+     ,[newName,newCity ,newAdd,newCountry,customerId],(error)=> {
+            if ((error)== undefined){
+                
+                res.send("customer "+customerId+" updated.");
+            }
+        });
+    }else
+    res.status(400).send("This id "+customerId  +" is not  existed  ")
+    });
+
+    //Delete order
+    app.delete("/orders/:orderId" ,function (req,res){
+        const orderId=parseInt( req.params.orderId);
+    
+        pool.query("delete from order_items  where  order_id=$1"
+     ,[orderId],(error)=> {
+            if ((error)== undefined){
+                pool.query("delete from orders  where  id=$1"
+     ,[orderId],(error)=> {
+            if ((error)== undefined){
+                
+                res.send("order "+orderId+"  is deleted.");
+            }
+        });
+             
+            }
+        });
+        
+    });
+
+    //get all orders of specific customer
+    app.get("/customers/:customerId/orders", function(req, res) {
+        const cudtomerId=parseInt(req.params.customerId);
+        pool.query(' select o.order_reference,o.order_date,p.product_name,oi.quantity,s.supplier_name,pa.unit_price  '+
+        ' from orders o  join order_items oi  on(o.id=oi.order_id)'+
+        ' join product_availability pa on(oi.product_id=pa.prod_id and oi.supplier_id=pa.supp_id)'+
+        ' join products p  on(p.id=pa.prod_id)'+
+        ' join suppliers s on(s.id=pa.supp_id)'+
+
+        ' where o.customer_id=$1 ',[cudtomerId], (error, result) => {
+            res.json(result.rows);
+        });
+    });
+
+app.listen(3000, function() {
+    console.log("Server is listening on port 3000. Ready to accept requests!");
+});
```

Comment:
You have missed out the last but one task - to delete a customer.
Also, ideally you should handle the error case in each function and also return an appropriate status code for the successful responses and the error responses.